bentoml/_internal/configuration/containers.py

```diff
@@ -194,9 +220,82 @@ def _is_ip_address(addr: str) -> bool:
     }
 )
 
-_WARNING_MESSAGE = (
-    "field 'api_server.%s' is deprecated and has been renamed to 'api_server.http.%s'"
-)
+_WARNING_MESSAGE = "field '%s.%s' is deprecated and has been renamed to '%s'"
+
+
+def v1_to_v2_migration(config_merger: Merger, override_config: dict[str, t.Any]):
+    # api_server compat
+    if "api_server" in override_config:
+        deprecated = ["port", "host", "max_request_size", "cors"]
+        api_server_config = override_config["api_server"]
+        # check if user are using older configuration
+        if "http" not in api_server_config:
+            api_server_config["http"] = {}
+        # compat layer
+        for field in deprecated:
+            if field in api_server_config:
+                old_field = api_server_config.pop(field)
+                api_server_config["http"][field] = old_field
+                logger.warning(
+                    _WARNING_MESSAGE, "api_server", field, f"api_server.http.{field}"
+                )
+        config_merger.merge(override_config["api_server"], api_server_config)
+        assert all(
+            key not in override_config["api_server"]
+            for key in ["cors", "max_request_size", "host", "port"]
+        )
+    # tracing compat
+    if "tracing" in override_config:
```

Comment:
Maybe a utility function `rename_field` might be useful? e.g. `rename_field(config, "zipkin.url", "zipkin.endpoint")` or something? I think it would make this easier to read.

Reply:
sg.